packages/data-api/src/SqlQuery.js

```diff
@@ -6,22 +6,51 @@
 export class SqlQuery {
   static parameteriseArray = arr => `(${arr.map(() => '?').join(',')})`;
 
-  constructor(baseQuery, baseParameters) {
+  static parameteriseValues = arr => `VALUES (${arr.map(() => `?`).join('), (')})`;
+
+  constructor(baseQuery, baseParameters = []) {
     this.query = baseQuery;
     this.parameters = baseParameters;
     this.orderByClause = null;
+    this.hasWhereClause = false;
+  }
+
+  loggableQuery() {
+    const replacementIterator = this.parameters
+      .map(param => param.replace(/'/g, "''"))
+      [Symbol.iterator]();
+    return this.query.replace(/\?/g, () => `'${replacementIterator.next().value}'`);
```

Comment:
This may be one of the more complex pieces of javascript I've ever written... purely for the purposes of logging 😂